update docs to include cpu/memory defaults for restic

a-mccarthy opened the pull request:
Signed-off-by: Abigail McCarthy <mabigail@vmware.com>

a-mccarthy commented:
@stephbman and @nrb heres the PR for doc updates to the CPU/Memory defaults for restic

nrb commented on site/docs/main/customize-installation.md:
```diff
@@ -95,10 +97,108 @@ velero client config set features=
+
+### Install with custom resource requests and limits
+
+You can customize these resource requests and limit when you first install using the [velero install][6] CLI command.
```
For other reviewers: I tested updating pod requests/limits with `velero install`, and it does not work. `kubectl edit` is required.

nrb commented on site/docs/main/customize-installation.md:
```diff
@@ -95,10 +97,108 @@ velero client config set features=
+
+### Update resource requests and limits after install
+
+After installation you can adjust the resource requests and limits in the Velero Deployment spec restic DeamonSet spec, if you are using the restic integration.
```
Should this read (emphasis added)

> in the Velero Deployment spec _or_ restic DeamonSet spec

nrb replied:
That's probably easier to copy/paste, yes.

nrb commented on site/docs/main/customize-installation.md:
```diff
@@ -304,4 +404,4 @@ If you get an error like `complete:13: command not found: compdef`, then add the
 [9]: self-signed-certificates.md
 [10]: csi.md
-[11]: https://github.com/vmware-tanzu/velero/blob/main/pkg/apis/velero/v1/constants.go
+[11]: https://github.com/vmware-tanzu/velero/blob/master/pkg/apis/velero/v1/constants.go
```
Looks like this got changed back to `master`

nrb commented on site/docs/main/restic.md:
```diff
@@ -1,40 +1,43 @@
-We integrated restic with Velero so that users have an out-of-the-box solution for backing up and restoring almost any type of Kubernetes
-volume*. This is a new capability for Velero, not a replacement for existing functionality. If you're running on AWS, and
-taking EBS snapshots as part of your regular Velero backups, there's no need to switch to using restic. However, if you've
+The restic intergation was added to give you an out-of-the-box solution for backing up and restoring almost any type of Kubernetes volume. This is a new capability for Velero, not a replacement for existing functionality. If you're running on AWS, and taking EBS snapshots as part of your regular Velero backups, there's no need to switch to using restic. However, if you've
```
`intergation` -> `integration`

ashish-amarnath commented on site/docs/main/restic.md:
```diff
@@ -1,40 +1,43 @@
 **RancherOS**
 
-The host path for volumes is not `/var/lib/kubelet/pods`, rather it is `/opt/rke/var/lib/kubelet/pods`
+Update the host path for volumes from `/var/lib/kubelet/pods` to `/opt/rke/var/lib/kubelet/pods`.
```
We should include that this change has to be made to the restic daemon set in the velero namespace.

a-mccarthy replied:
@ashish-amarnath, so to clarify, someone would need to run the velero install command and then do the steps in this section. I was talking to @nrb about this, and some of the information in this section a user should do before installing. Are their things in the `Configure restic DaemonSet spec` section that should be done before install?

ashish-amarnath replied:
The restic daemon set is part of the installation when `--use-restic` is supplied to the `velero install command`
So prior to installation, there will not be a restic daemon set.

nrb replied:
@a-mccarthy Ashish is correct here and I was wrong when we spoke over Zoom. Updating the `hostPath` is done _after_ installation.

a-mccarthy replied:
@nrb and @ashish-amarnath i've update this back to the original so the deamonset configurations steps are after the velero install command

ashish-amarnath commented on site/docs/main/restic.md:
````diff
@@ -48,17 +51,19 @@ hostPath:
   path: /opt/rke/var/lib/kubelet/pods
 ```
 
+Now you can [install Velero](#install-restic) with the restic integration.
````
Installing Velero again will overwrite the update that was just made.

ashish-amarnath commented on site/docs/main/restic.md:
```diff
@@ -48,17 +51,19 @@ hostPath:
 **OpenShift**
 
-The restic containers should be running in a `privileged` mode to be able to mount the correct hostpath to pods volumes.
+To be able to mount the correct hostpath to pods volumes, run the restic containers in a `privileged` mode.
```
suggest:
```
To mount the correct hostpath to pod volumes, run the restic containers in `privileged` mode.
```

a-mccarthy replied:
would it be more accurate to say `restic pods` instead of `restic containers`? or do you mean the containers, and not a pod in this case?

a-mccarthy replied:
@ashish-amarnath see above q :)

ashish-amarnath replied:
sorry for the latency :)
Yes, restic pods would be accurate.

a-mccarthy replied:
thanks, i updated this to say pods to be consistent.

ashish-amarnath commented on site/docs/main/restic.md:
````diff
@@ -133,7 +138,9 @@ oc annotate namespace <velero namespace> openshift.io/node-selector=""
 ```
 
-**Enterprise PKS**
+Now you can [install Velero](#install-restic) with the restic integration.
````
same comment as L#54

ashish-amarnath commented on site/docs/main/restic.md:
````diff
@@ -155,8 +163,7 @@ kubectl patch storageclass/<YOUR_AZURE_FILE_STORAGE_CLASS_NAME> \
 ```
-
-You're now ready to use Velero with restic.
+Now you can [install Velero](#install-restic) with the restic integration.
````
same comment as L#54

a-mccarthy commented on site/docs/main/customize-installation.md:
```diff
@@ -10,6 +10,8 @@
     - [Enable client side features](#enable-client-side-features)
   - [Customize resource requests and limits](#customize-resource-requests-and-limits)
+    - [Install with custom resource requests and limits](install-with-custom-resource-requests-and-limits)
+    - [Update resource requests and limits after install](update-resource-requests-and-limits-after-install)
```
updated :)

a-mccarthy commented on site/docs/main/customize-installation.md:
```diff
@@ -95,10 +97,108 @@ velero client config set features=
+  <tr><th>Setting</th><th>Velero pod defaults</th><th>restic pod defaults (Velero 1.4.2 and later)</th></tr>
+  <tr><td>CPU request</td><td>500m</td><td>500m</td></tr>
+  <tr><td>Memory requests</td><td>128Mi</td><td>512Mi</td></tr>
+  <tr><td>CPU limit</td><td>1000m (1)</td><td>1000m (1)</td></tr>
```
updated to say 1 CPU, is that more clear?

ashish-amarnath replied:
Yes, that does. Sorry I didn't make that connection when I first looked at it.

a-mccarthy commented on site/docs/main/customize-installation.md:
```diff
@@ -95,10 +97,108 @@ velero client config set features=
+  --velero-pod-cpu-request <CPU_REQUEST>] \
+  --velero-pod-mem-request <MEMORY_REQUEST>] \
+  --velero-pod-cpu-limit <CPU_LIMIT>] \
+  --velero-pod-mem-limit <MEMORY_LIMIT>]
```
updated

ashish-amarnath commented on site/docs/main/restic.md:
```diff
@@ -1,40 +1,43 @@
-volume*. This is a new capability for Velero, not a replacement for existing functionality. If you're running on AWS, and
-taking EBS snapshots as part of your regular Velero backups, there's no need to switch to using restic. However, if you've
+The restic intergation was added to give you an out-of-the-box solution for backing up and restoring almost any type of Kubernetes volume. This is a new capability for Velero, not a replacement for existing functionality. If you're running on AWS, and taking EBS snapshots as part of your regular Velero backups, there's no need to switch to using restic. However, if you've
 been waiting for a snapshot plugin for your storage platform, or if you're using EFS, AzureFile, NFS, emptyDir,
```
Thank you!

nrb commented on site/docs/main/customize-installation.md:
````diff
@@ -95,10 +97,108 @@ velero client config set features=
+    kubectl edit deploy velero -n velero
+    ```
+
+1. Update the `spec.template.spec.containers.args.resources.limits` and `spec.template.spec.containers.args.resources.requests` values.
````
The `args` element here needs to be removed.

nrb commented on site/docs/main/customize-installation.md:
````diff
@@ -95,10 +97,108 @@ velero client config set features=
+    kubectl edit daemonset restic -n velero
+    ```
+
+1. Update the `spec.template.spec.containers.args.resources.limits` and `spec.template.spec.containers.args.resources.requests` values.
````
The `args` element here needs to be removed.

nrb commented on site/docs/main/customize-installation.md:
````diff
@@ -95,10 +97,108 @@ velero client config set features=
+  [--restic-pod-mem-limit <MEMORY_LIMIT>]
+```
+
+### Update resource requests and limits after install
````
Here's `kubectl patch` commands for the Velero `Deployment` and restic `Daemonset`:

`kubectl patch deployment velero -n velero --patch '{"spec":{"template":{"spec":{"containers":[{"name": "velero", "resources": {"limits":{"cpu": "4", "memory": "512Mi"}, "requests": {"cpu": "3", "memory": "256Mi"}}}]}}}}'`

`kubectl patch daemonset restic -n velero --patch '{"spec":{"template":{"spec":{"containers":[{"name": "restic", "resources": {"limits":{"cpu": "3", "memory": "4Gi"}, "requests": {"cpu": "1", "memory": "1Gi"}}}]}}}}'`

These values are samples. I tested them out against a running cluster and both applied cleanly.

a-mccarthy replied:
@nrb ok, i can update to use the patch commands instead of using the `kubectl edit`. These commands are long though, so most of them won't show to the reader without scrolling. Is there are way to break it up into different lines, like by using `/` at the end of the a line?

nrb replied:
@a-mccarthy Yeah, these can be broken with`\`, but _not_ within the actual JSON, the stuff inside the `'`, unfortunately. JSON's pretty picky about syntax, and that slash won't be valid.

I think it would look like this:

```
kubectl patch daemonset restic -n velero --patch \
'{"spec":{"template":{"spec":{"containers":[{"name": "restic", "resources": {"limits":{"cpu": "3", "memory": "4Gi"}, "requests": {"cpu": "1", "memory": "1Gi"}}}]}}}}'
```

nrb replied:
We can provide them as YAML files, but I think @ashish-amarnath's original goal was to provide a more compact command, though I may be wrong.

nrb commented on site/docs/main/customize-installation.md:
````diff
@@ -95,10 +97,76 @@ velero client config set features=
+
+```bash
+kubectl patch daemonset restic -n velero --patch \
+'{"spec":{"template":{"spec":{"containers":[{"name": "restic", "resources": {"limits":{"cpu": "1", "memory": "512Mi"}, "requests": {"cpu": "1", "memory": "1024Mi"}}}]}}}}'
````
Requests must be less than limits, so these values on memory should be swapped. Otherwise, the command works.

a-mccarthy replied:
whoops, this should be fixed now!

nrb commented on site/docs/main/customize-installation.md:
````diff
@@ -95,10 +97,76 @@ velero client config set features=
+Update the `spec.template.spec.containers.resources.limits` and `spec.template.spec.containers.resources.requests` values in the Velero deployment.
+
+```bash
+kubectl patch deployment velero -n velero --patch \
````
Requests must be less than limits, so these values on memory should be swapped. Otherwise, the command works.

a-mccarthy replied:
same as above

a-mccarthy commented:
@nrb and @ashish-amarnath thanks for the reviews! I believe i've addressed all the open comments. I also added the content to the 1.4 docs. Please let me know if there is anything else that should be done here :)
